packages/grafana-ui/src/components/PieChart/types.ts

```diff
@@ -1,17 +1,32 @@
-import { DataFrame, FieldConfigSource, FieldDisplay, InterpolateFunction, ReduceDataOptions } from '@grafana/data';
+import {
+  DataFrame,
+  EventBusWithSource,
+  FieldConfigSource,
+  FieldDisplay,
+  InterpolateFunction,
+  ReduceDataOptions,
+} from '@grafana/data';
 import { VizLegendOptions } from '..';
 
 export interface PieChartSvgProps {
   height: number;
   width: number;
   fieldDisplayValues: FieldDisplay[];
   pieType: PieChartType;
+  highlightedTitle?: string;
   displayLabels?: PieChartLabels[];
   useGradients?: boolean;
   onSeriesColorChange?: (label: string, color: string) => void;
 }
 
-export interface PieChartProps extends Omit<PieChartSvgProps, 'fieldDisplayValues'> {
+export interface PieChartProps {
+  height: number;
+  width: number;
+  pieType: PieChartType;
+  displayLabels?: PieChartLabels[];
+  useGradients?: boolean;
+  onSeriesColorChange?: (label: string, color: string) => void;
+  eventBus?: EventBusWithSource;
```

Comment:
Not sure PieChart should accept eventBus. I would expect it to rather take in event callbacks that could be configured outside to use or not the event bus, i.e. onArcHover, onArcClick etc. Then the event bus is an implementation detail configured from the panel.

Reply:
Yeah this is the question about what level we should put the EventBus on. It's hard for me to reason about it without seeing the use cases for some other visualizations.

Reply:
the problem with that is that then PieChart needs prop for highlighted series, and everyone using PieChart who wants legend hover highlight will have to do this evenbus wiring